Sources/MapboxNavigation/UIView.swift

```diff
@@ -122,7 +122,11 @@ extension UIView {
         layer.render(in:currentContext)
         let image = UIGraphicsGetImageFromCurrentImageContext()
         UIGraphicsEndImageContext()
-        return image
+
+        // Check the transform property to see if the view was flipped.
+        // If it was then we need to apply a flip transform here as well since layer.render() ignores the view's transform when it is rendered
+        let isFlipped = transform.a == -1
+        return isFlipped ? image?.withHorizontallyFlippedOrientation() : image
```

Comment:
The annoying thing with this is that the transform isn't set *until* the render call is made, since it is set by the `draw(rect:)` method of the ManeuverView. If we want to use the same context to draw the flipped image into, we'll need to check for the flip after the image is rendered, then clear the context, apply the transform, and then re-render the view.
It might still be best to do this for complete correctness in all transform cases, but it's unfortunate. Do you feel like I should still go this route?

If so, then I think the code basically looks like this:

var imageRepresentation: UIImage? {
        let size = CGSize(width: frame.size.width, height: frame.size.height)
        UIGraphicsBeginImageContextWithOptions(size, false, (window?.screen ?? UIScreen.main).scale)
        guard let currentContext = UIGraphicsGetCurrentContext() else { return nil }

        layer.render(in:currentContext)

        // Check the transform property to see if the view was flipped.
        // If it was then we need to apply a flip transform here as well since layer.render() ignores the view's transform when it is rendered
        if transform.a != 1 {
            currentContext.clear(CGRect(origin: .zero, size: size))
            let flipTransform = CGAffineTransform(scaleX: -1, y: 1)
            currentContext.concatenate(flipTransform.translatedBy(x: -size.width, y: 0)) // flip the context and translate it to move it back into view.
            layer.render(in:currentContext)
        }
        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }

Reply:
I could also create a second image context, flip it, then blit the pixels from the first one into it, but I'm not sure if that's really better. Maybe it saves on rendering cost, but I doubt it's generally very expensive to do those renders.

Reply:
We could move the intended transform into a helper method that this method can access independently of actually drawing. That would be similar in spirit to #2809.

Reply:
Actually, we can make this tail work. I’m just concerned any discrepancy will slip through the cracks because this is such a convenient method to use in e.g. CarPlay, but it’s only a theoretical concern at this point.